torch/_export/serde/serialize.py

```diff
@@ -636,45 +689,86 @@ def serialize(self, exported_program: ep.ExportedProgram) -> Tuple[ExportedProgr
 class GraphModuleDeserializer:
     def __init__(self):
         self.serialized_name_to_node: Dict[str, torch.fx.Node] = {}
-        self.serialized_name_to_meta: Dict[str, Union[FakeTensor, int, torch.SymInt, torch.SymBool]] = {}
+        self.serialized_name_to_meta: Dict[str, MetaType] = {}
         self.graph = torch.fx.Graph()
-        self.fake_tensor_mode = FakeTensorMode()
 
-    def deserialize_sym(
-        self, s: Union[SymInt, SymBool]
-    ) -> Union[int, torch.SymInt, bool, torch.SymBool]:
+    def deserialize_sym_int(self, s: SymInt) -> Union[int, torch.SymInt]:
         val = s.value
-        if s.type == "as_symbol":
-            raise NotImplementedError("TODO(angelayi)")
+        if s.type == "as_expr":
+            if val.expr_str in self.symbol_name_to_symbol:
+                sym = self.symbol_name_to_symbol[val.expr_str]
+            else:
+                sym = sympy.sympify(val.expr_str, locals=self.symbol_name_to_symbol)
+
+                if isinstance(sym, sympy.Symbol):
+                    self.symbol_name_to_symbol[val.expr_str] = sym
+
+                    if vr := self.symbol_name_to_range.get(val.expr_str):
+                        self.shape_env.var_to_range[sym] = symbolic_shapes.ValueRanges(
+                            vr.lower, vr.upper
+                        )
```

Comment:
I can't because the constrain API doesn't take in a symbol as input 😅 but it uses a symbol to set the range...it's easier if I just set it directly...